### PR TITLE
Remove ROCm backend

### PR DESCRIPTION
**Pull request by masterleinad**
Together with #3135, #3145 and #3147, this pull request removes the `ROCm` backend.

### CODE REVIEW

**Comment by DavidPoliakoff** [comment]
:+1: :+1: :+1: :+1: :+1: :+1: :+1: :+1: :+1: :+1: :+1: :+1: :+1: :+1: :+1: :+1:

**Review by dalg24 — DISMISSED** [review]
Do not merge before @Rombur approves

**Review by crtrott — APPROVED** [review]

**Comment by dalg24** [comment]
This can be rebased and merged now.

**Comment by masterleinad** [comment]
Rebased.

File: containers/unit_tests/TestScatterView.hpp
Comment (dalg24) [inline containers/unit_tests/TestScatterView.hpp:726]: Why is this part of this PR

Comment (masterleinad) [reply containers/unit_tests/TestScatterView.hpp:726]: While removing `ROCm` with its tests I was checking if we were missing any tests in related files. Let me open a separate pull request with these changes.

Comment (masterleinad) [reply containers/unit_tests/TestScatterView.hpp]: See #3336. Removed here.

**Review by dalg24 — APPROVED** [review]

**Pull request merged**
